File: javascript/conduits/sendMessage.js
import { PWRWallet } from "@pwrjs/core";
import dotenv from 'dotenv';
dotenv.config();

// Setting up your wallet in the SDK
const privateKey = process.env.PRIVATE_KEY;
const wallet = new PWRWallet(privateKey);

async function sendMessage() {
    const obj = { message: "please send me pwr" };
    const data = Buffer.from(JSON.stringify(obj), 'utf8');
    const vmId = 123;

    const res = await wallet.sendVMDataTxn(vmId, data);
    console.log(res.transactionHash);
}
sendMessage();

File: rust/examples/messages_dapp/send_message.rs
use pwr_rs::Wallet;
use dotenvy::dotenv;
use std::env;
use serde_json::json;

async fn send_message() {
    dotenv().ok();
    // Setting up your wallet in the SDK
    let private_key = env::var("PRIVATE_KEY").unwrap();
    let wallet = Wallet::from_hex(&private_key).unwrap();

    let obj = json!({ "message": "lfg" });
    let data = serde_json::to_vec(&obj).unwrap(); // Serialize to JSON bytes
    let vm_id = 1234;
    let res = wallet.send_vm_data(vm_id, data).await;
    println!("{}", res);
}

#[tokio::main]
pub async fn main() {
    send_message().await;
}